src/osp/Active/SysPrefabInit.cpp

```diff
@@ -41,35 +42,43 @@ using osp::phys::EShape;
 namespace osp::active
 {
 
-void SysPrefabInit::init_hierarchy(
-        ACtxPrefabInit const&               rPrefabInit,
+void SysPrefabInit::add_to_subtree(
+        TmpPrefabInitBasic const&           basic,
+        ArrayView<ActiveEnt const>          ents,
         Resources const&                    rResources,
-        acomp_storage_t<ACompHierarchy>&    rHier) noexcept
+        SubtreeBuilder&                     bldPrefab) noexcept
 {
-    auto itPfEnts = std::begin(rPrefabInit.m_ents);
+    auto const &rImportData = rResources.data_get<osp::ImporterData const>(
+            gc_importer, basic.m_importerRes);
+    auto const &rPrefabData = rResources.data_get<osp::Prefabs const>(
+            gc_importer, basic.m_importerRes);
 
-    for (TmpPrefabInitBasic const& rPfBasic : rPrefabInit.m_basic)
-    {
-        auto const &rPrefabData = rResources.data_get<osp::Prefabs const>(
-                gc_importer, rPfBasic.m_importerRes);
+    auto const objs     = rPrefabData.m_prefabs[basic.m_prefabId];
+    auto const parents  = rPrefabData.m_prefabParents[basic.m_prefabId];
```

Comment:
i assume these are cheap to copy?

Reply:
These are just spans. I often use auto for any kind of range, though I guess it's possible to return some larger types in many other cases. Would it be good to use const-ref instead?